DuckDuckGo/VoiceSearchHelper.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  VoiceSearchHelper.swift
+//  DuckDuckGo
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+
+protocol VoiceSearchHelperProtocol {
+    var isSpeechRecognizerAvailable: Bool { get }
+    var privacyAlertWasConfirmed: Bool { get }
+    func markPrivacyAlertAsConfirmed()
+}
+
+class VoiceSearchHelper: VoiceSearchHelperProtocol {
+    private(set) var isSpeechRecognizerAvailable: Bool = false
+    private var variantManager: VariantManager?
+
+    @UserDefaultsWrapper(key: .voiceSearchPrivacyAlertWasConfirmed, defaultValue: false)
+    private(set) var privacyAlertWasConfirmed: Bool
+    
+    init(_ variantManager: VariantManager? = nil) {
+        self.variantManager = variantManager
+        
+        updateFlag()
+        
+        NotificationCenter.default.addObserver(forName: NSLocale.currentLocaleDidChangeNotification, object: nil, queue: nil) { [weak self] _ in
+            self?.updateFlag()
```

Comment:
@bwaresiak given that the app is restarted I was actually considering removing this (and forgot, so thanks for pointing it out), so my suggestion would be to remove this code because it's not actually doing anything useful unless Apple changes how the system behaves, since today this notification is only called once the user changes temperature unit, number format, etc. But not language which is what matters for this.

I'll remove this but let me know if you disagree.